### PR TITLE
Add pre-commit

### PR DESCRIPTION
**Pull request by PrasadJoshi12**
Fix issue #6

### CODE REVIEW

**Comment by PrasadJoshi12** [comment]
@jyejare Can you please review it? Please let me know if any check is missing :smile:

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+repos:
+  -   repo: https://github.com/pre-commit/pre-commit-hooks
+      rev: v2.3.0
```
Comment (jyejare) [inline .pre-commit-config.yaml:3]:
```suggestion
      rev: v4.0.1
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      hooks:
+      -   id: end-of-file-fixer
+      -   id: trailing-whitespace
+      -   id: check-yaml
```
Comment (jyejare) [inline .pre-commit-config.yaml:7]: Add `- id: debug-statements`

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  - repo: https://gitlab.com/PyCQA/flake8
+    rev: 3.8.3
```
Comment (jyejare) [inline .pre-commit-config.yaml:9]: Use Latest: 3.9.2

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    hooks:
+      -   id: flake8
+  - repo: https://github.com/psf/black
+    rev: 20.8b1
```
Comment (jyejare) [inline .pre-commit-config.yaml:13]: Use latest 22.3.0 !

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    hooks:
+      - id: black
+        args: [ --config=pyproject.toml ]
```
Comment (jyejare) [inline .pre-commit-config.yaml:16]: I think you dont need to explicitly mention the path, it should auto detect. Just remove the args and check !

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+[tool.black]
+line-length = 79
```
Comment (jyejare) [inline pyproject.toml:2]: Go upto 100 lines no worries !

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+
+[tool.isort]
+line_length = 79
```
Comment (jyejare) [inline pyproject.toml:5]: 100 ??

File: requirements.txt [hunk 8]
```diff
@@ -2,3 +2,4 @@ wrapanapi==3.5.10
 dynaconf==3.1.2
 wget==3.2
 click==7.1.2
+pre-commit==2.16.0
```
Comment (jyejare) [inline requirements.txt:5]: latest is 2.19.0

**Review by jyejare — CHANGES_REQUESTED** [review]
Do the required changes !

**Comment by jyejare** [comment]
@mshriver Great idea !

**Comment by jyejare** [comment]
Ping @PrasadJoshi12 !

File: requirements.txt (hunk 8 above)
Comment (PrasadJoshi12) [reply requirements.txt]: Fixed. Could you please review it again ? @jyejare

File: .pre-commit-config.yaml (hunk 1 above)
Comment (PrasadJoshi12) [reply .pre-commit-config.yaml]: Updated

File: .pre-commit-config.yaml (hunk 2 above)
Comment (PrasadJoshi12) [reply .pre-commit-config.yaml]: Added

File: pyproject.toml (hunk 6 above)
Comment (PrasadJoshi12) [reply pyproject.toml]: Changed to 100 lines

**Review by jyejare — APPROVED** [review]

**Comment by jyejare** [comment]
@PrasadJoshi12 ACKed and Merging now! 

I shall raise a follow-up PR enforcing the pre-commit changes along with the Github Action enforcing every PR to follow the pre-commit!

**Pull request merged**